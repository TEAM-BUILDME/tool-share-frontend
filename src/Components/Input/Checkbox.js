import React from 'react';

const Checkbox = ({value, text, onChange}) => {
	return (
		<div>
			<label>
			<input 
				type='checkbox' 
				value={value} 
				onChange={onChange}
			/>
				{text.toString()}
			</label>
		</div>
	);
};

export default Checkbox;
