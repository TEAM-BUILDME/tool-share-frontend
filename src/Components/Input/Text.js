import React, { useState } from 'react';

const TextInput = () => {
	const [ input ] = useState('');
	const handleChange = (e) => {
		
	};
	return (
		<input type="text" onChange={handleChange} />
	);
}

export default TextInput;
