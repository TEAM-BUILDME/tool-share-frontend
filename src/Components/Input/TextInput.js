import React from 'react';

const TextInput = ({ name, placeholder, value, onChange }) => {
	const handleChange = (e) => { 
		onChange(e.target.value); 
	}

	return (
		<input 
			name={name} 
			type='text' 
			placeholder={placeholder} 
			value={value}
			onChange={handleChange} />
	);
}

export default TextInput;
