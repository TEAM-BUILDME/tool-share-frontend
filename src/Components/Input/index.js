import React, { useState } from 'react';

const Input = () => {
	const [ input ] = useState('');
	const handleChange = (e) => {
		
	};
	return (
		<input type="text" onChange={handleChange} />
	);
}

export default Input;
