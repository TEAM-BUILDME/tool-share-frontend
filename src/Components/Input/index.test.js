import React from 'react';
import { shallow } from 'enzyme';

import Input from './index.js'

describe('Input', () => {
	it('should render an input element', () => {
		const component  = shallow(<Input />);
		expect(component.getElements()).toMatchSnapshot();
	});
	it('should create an entry in component state', () => {
		const component = shallow(<Input />);
		const form = component.find('input');
		form.props().onChange({target: {
			name: 'someName',
			value: 'someValue'
		}});
		expect(component.state.input).toBeDefined();
	});
});
