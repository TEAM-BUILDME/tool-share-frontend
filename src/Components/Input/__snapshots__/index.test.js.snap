// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render an input element 1`] = `
Array [
  <input
    onChange={[Function]}
    type="text"
  />,
]
`;
